Core/Core/Api/GraphQL/GraphQLHttpClientExtensions.cs

```diff
@@ -0,0 +1,47 @@
+using GraphQL;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Threading;
+using GraphQL.Client.Http;
+
+namespace Speckle.Core.Api.GraphQL;
+
+public static class GraphQLHttpClientExtensions
+{
+  /// <summary>
+  /// Gets the version of the current server. Useful for guarding against unsupported api calls on newer or older servers.
+  /// </summary>
+  /// <param name="cancellationToken">[Optional] defaults to an empty cancellation token</param>
+  /// <returns><see cref="Version"/> object excluding any strings (eg "2.7.2-alpha.6995" becomes "2.7.2.6995")</returns>
+  /// <exception cref="SpeckleGraphQLException"></exception>
+  public static async Task<System.Version> GetServerVersion(
+    this GraphQLHttpClient client,
+    CancellationToken cancellationToken = default
+  )
```

Comment:
Why an extension method?

Reply:
I like extensions methods. They are reusable, discoverable with intellisense, and organized so we'll never end up with massive utils files. I didn't want to add it as a method on line 850 of AccountManager so this is what I came up with, but I'm happy to change it if you have something else in mind.

Reply:
yeah i do agree 850 lines seems like enough. But extension methods were really created to extend a class you don't have access to (i.e from a dependency), this seems a bit of a misuse, but I accept your reasoning :)